data_model/src/events/pipeline.rs

```diff
@@ -29,48 +29,65 @@ pub use crate::transaction::RejectionReason as PipelineRejectionReason;
     Deserialize,
 )]
 pub struct EventFilter {
-    /// Filter by Entity if `Some`, if `None` all entities are accepted.
-    pub entity: Option<EntityType>,
-    /// Filter by Hash if `Some`, if `None` all hashes are accepted.
+    /// If `Some::<EntityKind>` filters by the [`EntityKind`]. If `None` accepts all the [`EntityKind`].
+    pub entity_kind: Option<EntityKind>,
+    /// If `Some::<StatusKind>` filters by the [`StatusKind`]. If `None` accepts all the [`StatusKind`].
+    pub status_kind: Option<StatusKind>,
+    /// If `Some::<Hash>` filters by the [`Hash`]. If `None` accepts all the [`Hash`].
     pub hash: Option<Hash>,
 }
 
 impl EventFilter {
-    /// Filter by entity
-    pub const fn by_entity(entity: EntityType) -> EventFilter {
-        EventFilter {
-            entity: Some(entity),
-            hash: None,
-        }
+    /// Construct [`EventFilter`].
+    #[must_use]
+    #[inline]
+    pub fn new() -> Self {
```

Comment:
Hm..., you can't have it `const` because `Default` isn't const.
I'll just leave some interesting upcomming features here: https://stackoverflow.com/questions/70441495/what-is-impl-const-in-rust